util/str_utils.py

```diff
@@ -0,0 +1,26 @@
+import string
+
+PUNCTUATION_TRANS = str.maketrans(string.punctuation, ' ' * len(string.punctuation))
+WHITESPACE_TRANS = str.maketrans(string.whitespace, ' ' * len(string.whitespace))
+
+
+def preprocess_text(text):
+    """Method for pre-processing the given response text.
+    It:
+    * changes to lowercase
+    * removes trailing and leading spaces
+    * removes all punctuations
+    * removes double spaces
+
+    :param text: the text to be cleaned
+    :return: cleaned text
+    """
+    text = text.strip().lower()
+
+    text = text.translate(PUNCTUATION_TRANS)
+    text = text.translate(WHITESPACE_TRANS)
+
+    while '  ' in text:
+        text = text.replace('  ', ' ')
```

Comment:
I'm not sure if I placed the last comment in the proper place - this while loop seems like an overkill and `re.sub(' +', '', text)` seems simplier. In the previous comment regarding this you mentioned the efficiency of translate, but my comment was about the while loop.

Reply:
`re.sub(' +', ' ', text)` is simplier but not the most efficient one (takes 3x more time than while loop). Anyways, I don't think it'll impact our performance much, so I'll change it.

Fixed in 625e2c754e39226e83ff1895cda4e16c728fe7fe

Reply:
Didn't know the efficiency difference is that big, never tested that nor seen any tests. Shouldn't be a bottleneck, but worth remembering.